src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/probation/DeliusApiClient.kt

```diff
@@ -51,36 +53,29 @@ class DeliusApiClient(@Qualifier("oauthDeliusApiClient") val communityApiClient:
       ?: error("Unexpected null response from API")
   }
 
-  fun getStaffEmails(crns: List<String>): List<StaffEmail> {
-    if (crns.isEmpty()) return emptyList()
-
-    val batchedCrns = crns.chunked(STAFF_EMAIL_BATCH)
-    val results = batchedCrns.map { batch ->
+  fun getStaffEmails(crns: List<String>) =
+    batchRequests(STAFF_EMAIL_BATCH, crns) { batch ->
       communityApiClient
         .post()
         .uri("/probation-case/responsible-community-manager")
         .bodyValue(batch)
         .accept(MediaType.APPLICATION_JSON)
         .retrieve()
         .bodyToMono(typeReference<List<StaffEmail>>())
-        .block() ?: error("Unexpected null response from API")
```

Comment:
How come we don't want to through an error here anymore? I guess it would just throw an error anyway if an error occurred?

Reply:
It was just to make it consistent with the others - the batching function handles nulls by coercing them into an empty list.

I think it makes sense to handle these consistently as there's no reason why the behaviour should differ based on the endpoint. I went with coerce to null rather than error, as that was the most common pattern - there is an argument to consistently error if we encounter nulls though.

Reply:
Around this point, there are a couple of places where we propagate 404's to 404 rather 500 (eg in the `HdcApiClient`) - would be good to get consistent on that as well throughout our codebases. In terms of nulls, how would we ever know then if we coerce whether a genuine error with null data has occurred?